### PR TITLE
.travis.yml: Disable install on unsupported

### PR DESCRIPTION
**Pull request by jayvdb**
Unsupported versions do not need requirements
pre-installed.  This also prevents version
constraints in dependencies from breaking the job.

Closes https://github.com/coala/coala/issues/5636

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by jayvdb** [comment]
Example build with unsupported jobs: https://travis-ci.org/jayvdb/coala/builds/415949300

File: .travis.yml [hunk 1]
```diff
@@ -12,15 +12,19 @@ stages:
   - name: unsupported
     if: branch = master AND type = push
 
+.disable_global: &disable_global
```
Comment (jayvdb) [inline .travis.yml]: note this convention is being used in lots of coala repos now
https://github.com/coala/coala-bears/blob/master/.travis.yml#L9

File: .travis.yml [hunk 2]
```diff
@@ -38,12 +42,15 @@ jobs:
...
       script: .misc/check_unsupported.sh
     - python: 3.3
+      <<: *disable_global
       stage: unsupported
       script: .misc/check_unsupported.sh
     - python: 3.4.3
+      <<: *disable_global
       stage: unsupported
       script: .misc/check_unsupported.sh
     - python: 3.6
```
Comment (ksdme) [inline .travis.yml]: To make this section more intuitive and readable, can we consider merging `disable_global` in this section too and then overriding install?

Comment (ksdme) [reply .travis.yml]: It is also partially possible to do this in https://github.com/coala/coala-bears/blob/master/.travis.yml#L62.

Comment (jayvdb) [reply .travis.yml]: Feel free to create a newcomer issue about the bears one; explain it fairly clearly, so domain knowledge isnt needed - only expect yaml skills to complete the task.

**Comment by ksdme** [comment]
LGTM.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5705/commits/966e1140f056cc91f3d5e766e23fba6990497b3e

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**